timeseries/src/autogluon/timeseries/models/abstract/abstract_timeseries_model.py

```diff
@@ -87,9 +87,10 @@ class AbstractTimeSeriesModel(AbstractModel):
     _preprocess_nonadaptive = None
     _preprocess_set_features = None
 
-    supports_known_covariates: bool = False
-    supports_past_covariates: bool = False
-    supports_static_features: bool = False
+    _supports_known_covariates: bool = False
+    _supports_past_covariates: bool = False
+    _supports_static_features: bool = False
+    _supports_cat_covariates: bool = False
```

Comment:
This attribute is only relevant for GluonTS models. Should we keep it there without the underscore (and maybe add a comment)?

Reply:
If we can imagine it being relevant for other models I would rather keep it at the top level?

Reply:
I feel that even the `supports_cat_covariates` flag for GluonTS models is kind of a hack because it's unclear if this applies to known, past, or both types of covariates. It currently happens that the only model where this is set to `True` (TFT) is also the only GluonTS model that supports past covariates, but this doesn't have to be true in general. 

I would prefer to keep it limited to GluonTS until we at least explore other backends for DL models.

Reply:
Got it. back it goes.

Reply:
I sent it back to gluonts, but kept it as a property to keep it private and consistent.